xarray/coding/times.py

```diff
@@ -181,6 +181,11 @@ def _decode_datetime_with_pandas(flat_num_dates, units, calendar):
         # strings, in which case we fall back to using cftime
         raise OutOfBoundsDatetime
 
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", "invalid value encountered", RuntimeWarning)
+        pd.to_timedelta(flat_num_dates.min(), delta) + ref_date
+        pd.to_timedelta(flat_num_dates.max(), delta) + ref_date
```

Comment:
I think we could probably get away with something like this now that the pandas issue (pandas-dev/pandas#14068) as been fixed:
```suggestion
        pd.to_timedelta(flat_num_dates.min(), delta)
        pd.to_timedelta(flat_num_dates.max(), delta)
```
Now we just want to make sure we raise an `OverflowError` if at least one of the values we eventually want to cast to a timedelta is too largely negative or positive, though it probably does not hurt to keep the `ref_date` addition in there.

For context, I think with the pandas issue we used to have to also guard against silent overflow in the last line of this function (e.g. if the timedeltas were valid, but were large enough when coupled to the reference date to produce out of bounds times):
https://github.com/pydata/xarray/blob/a6f51c680f4e4c3ed5101b9c1111f0b94d28a781/xarray/coding/times.py#L196-L200

Reply:
Ah ok nice got it, will leave as is then